corehq/apps/hqwebapp/views.py

```diff
@@ -393,6 +398,43 @@ def logout(req):
 def retrieve_download(req, domain, download_id, template="hqwebapp/file_download.html"):
     return soil_views.retrieve_download(req, download_id, template)
 
+
+def dropbox_next_url(request, download_id):
+    return request.META.get('HTTP_REFERER', '/')
+
+
+@require_dropbox_session(next_url=dropbox_next_url)
+def dropbox_upload(request, download_id):
+    download = DownloadBase.get(download_id)
+    if download is None:
+        logging.error("Download file request for expired/nonexistent file requested")
+        raise Http404
+    else:
+        filename = download.get_filename()
```

Comment:
i take that back, the multimedia specifies that it should be on the NFS drive (https://github.com/dimagi/commcare-hq/blob/dropbox/corehq/apps/hqmedia/tasks.py#L117). but still am not convinced that it's the responsibility of the dropbox upload to delete the file or ensure that the file is on the NFS drive. whatever portion of the code is creating the file using DownloadBase should make that the file is on the NFS. let me know if you think that makes sense